CHANGELOG.md

```diff
@@ -4,6 +4,25 @@
 <!--- Distributed under the Boost Software License, Version 1.0. (See accompanying -->
 <!--- file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt) -->
 
+## 0.4.0 (2022-05-04)
+
+### New features
+
+- Annotations are now inherited from the scheduling task instead of the spawning task for `transfer`. ([#172](https://github.com/pika-org/pika/pull/142))
```

Comment:
In the previous release notes I've linked to PRs rather than issues because they typically contain more information about the actual change (but also links to the issue), rather than speculation about what might be needed. I'm not set on this though, and we could link to issues if you think it's nicer. I suppose the no-compromise solution would be to link to both (if available).

Whatever we settle on, the numbers need to be fixed in the link (142 is another issue).

What do you think?

Reply:
I was thinking the issue could be found from the PR, but I don't mind putting both either
Will fix the number!

Reply:
I changed it to 188 i.e. the PR, let me know if you also want me to link the issue